beamer/util.py

```diff
@@ -176,10 +176,14 @@ def is_valid_pair(
         source_token_address: ChecksumAddress,
         target_chain_id: ChainId,
         target_token_address: ChecksumAddress,
+        current_target_chain_id: ChainId,
     ) -> bool:
         source_token = source_chain_id, source_token_address
         target_token = target_chain_id, target_token_address
-        return target_token in self._tokens.get(source_token, frozenset())
+        return (
+            target_token in self._tokens.get(source_token, frozenset())
+            and current_target_chain_id == target_chain_id
```

Comment:
Shouldn't the test also include a check against the source chain ID?
What happens if the agent is configured for (Op -> Boba) but the source chain is Arbitrum? 
In this case it would return true wouldnt it?